### DISCUSSION TITLE
DBeaver CE Build: Missing JRE Components in App Image

### DISCUSSION THREAD

**Discussion by edespino** [dbeaver/dbeaver#36086] [DBeaver Development]
**Environment**
- OS: Ubuntu 22.04
- Building: DBeaver Community Edition from source
- Build Result: Successfully creates DBeaver.app for both x86_64 and aarch64 architectures

**Issue**
After successfully building DBeaver CE, I've encountered an issue with the JRE integration in the final app images. Here are the specific details:

**Build Output Structure**
```
./product/community/target/products/org.jkiss.dbeaver.core.product/macosx/cocoa/x86_64/DBeaver.app
./product/community/target/products/org.jkiss.dbeaver.core.product/macosx/cocoa/aarch64/DBeaver.app
```

**Missing JRE Components**

- The `DBeaver.app/Contents/Eclipse/dbeaver.ini` files reference `libjli.dylib` in the following locations:

  - x86_64: `../Eclipse/jre/Contents/Home/lib/libjli.dylib`
  - aarch64: `../Eclipse/jre/Contents/Home/lib/libjli.dylib`

- A search for `libjli.dylib` in the product directory returns no results:

```
find product/community/target/products/org.jkiss.dbeaver.core.product -name libjli.dylib
# No results found
```

**Questions**

1. Is there an additional build step needed to include the OpenJDK JRE contents in the app image?
2. Should the build process automatically package the required JRE components, or is this a manual post-build step?
3. Are there specific build configuration parameters needed to ensure the JRE is properly bundled for both x86_64 and aarch64 architectures?
4. Is there a difference in how the JRE should be handled between the two architectures, given the slight difference in the `dbeaver.ini` paths?

**Additional Context**

- The build process completes without errors
- Both architecture variants of the app are created successfully
- The JRE directory structure referenced in `dbeaver.ini` is not present in either build
- The issue affects both x86_64 and aarch64 builds

Any guidance on resolving this JRE packaging issue would be greatly appreciated.

**Comment by edespino** [comment]
Any chance someone can help with my questions? Thank you in advance.

**Comment by chirontt** [comment]
I think when you build DBeaver from source, the JRE is not included in the resulting build artifacts, but the JRE is included in the [official releases](https://github.com/dbeaver/dbeaver/releases), hence the above `-vm` entry in `dbeaver.ini` i.e.:
```
-vm
../Eclipse/jre/Contents/Home/lib/libjli.dylib
```
would work for the official releases. To make it work when you build it from source, just manually delete the above `-vm` lines from the resulting `dbeaver.ini` file.

**Reply by edespino** [reply]
Cool ... I'm new to the community and was hoping there would be a way to build a version that would simply work without any editing. Apparently that isn't something an open source user can do. Regardless, thank you for your input. You are appreciated.

**Reply by chirontt** [reply]
If you want, you can remove the following line from the `product/community/DBeaver.product` file:
https://github.com/dbeaver/dbeaver/blob/fe2710f9fde91b7ddb4cfeb72d45c28d50b2c407/product/community/DBeaver.product#L13
then your resulting build will work without any further editing.

And I would suggest you submit a PR for that change. (I don't use Mac so I wouldn't submit the PR myself.)
